api/src/main/java/javax/observer/collection/CollectionChangeEvent.java

```diff
@@ -1,21 +1,69 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
 package javax.observer.collection;
 
-import java.util.Collections;
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
+import java.util.Collection;
 
 /**
- * TODO
- * @author Hendrik Ebbers
+ * 
+ * @param <T> element type of the collection
+ * @param <OC> observed collection type
+ * @param <CC> collection change type
+ * 
+ * @author Michael Hoffer (info@michaelhoffer.de)
  */
-public interface CollectionChangeEvent<E, C extends ObservableCollection<E, C, ?>, V extends CollectionChange<E, C>> {
+public interface CollectionChangeEvent<T, OC extends Collection<T>, CC extends CollectionChange<T>> {
```

Comment:
We have a problem here:

 ```java
OC extends ObservableCollection<T>
```
doesn't work, since the event could come from a collection that is not directly the observable. Just tried to change this in my implementation and things broke apart. 

When deciding whether the source type should be the `CollectionObservable` or the actual `Collection`, I think the collection is more important and useful.

So I'm skeptical about this change.